src/v/cluster/partition_balancer_planner.cc

```diff
@@ -206,9 +212,7 @@ std::pair<uint64_t, uint64_t> partition_balancer_planner::get_node_bytes_info(
                 total_free = reclaimable_size - overage;
             }
         }
-        // Clamp the free space to be conservative and account for any error
-        // from space monitoring logic.
-        total_free = std::min(total_free, node_state.data_disk.free);
```

Comment:
I wonder if the original intent here was to clamp to `data_disk.total` or `data_disk.free + reclaimable_size` or somesuch

Reply:
cc @bharathv if you recall the intent here?

Reply:
yikes.. this is just wrong.. I'm trying to remember.. 

I _think_ it has something to do with space management estimates being inaccurate in the first few health reports. IIRC space management computes these estimates (current/reclaimable) in an async way. So right after a broker boots up, these values may be under reported because it hasn't fully caught up with the on-disk state. That means that is a risk of overestimating free_space here by the balancer and it can schedule some heavy moves right away. So the idea was to clamp it to data_disk.free which is almost realtime reported by the OS. Obviously this logic breaks in a happy path when all the values are accurate and it doesn't factor in reclaimable space.

This was done in the last moment without tests, so I'm not surprised this crept in.

Reply:
Thanks! The problem makes sense, you can imagine `current`/`reclaimable` being transiently near 0 as partitions have yet to be registered, while target is 80% of disk capacity, and the balancer would think the node has a ton of space. I think based on that it `data_disk.free + reclaimable_size` seems like a reasonable value to clamp down to. When booting up we'd still underestimate the amount of free space, but as `reclaimable` becomes more accurate, so would this estimate. @mmaslankaprv wdyt?

Alternatively, perhaps we should treat these target/current/reclaimable as the truth, but not report them if we're still replaying the controller log / haven't yet accounted for the partitions we expect to be on disk.

Reply:
> I think based on that data_disk.free + reclaimable_size seems like a reasonable value to clamp down to.

that sounds reasonable to me, particularly when current/reclaimable are inaccurate.

Reply:
@mmaslankaprv could you also make this change? Seems like there's some important rationale behind having some form of clamping